src/VersionTool/Program.cs

```diff
@@ -225,13 +253,324 @@ private static int OnList(CommandOption pathOption)
+                        {
+                            repo.Packages.UnionWith(Directory.EnumerateDirectories(repoSubDir).Select(path => new DirectoryInfo(path).Name));
+                        }
+
+                        foreach (var projectJsonPath in Directory.GetFiles(repoSubDir, "project.json", SearchOption.AllDirectories))
+                        {
+                            AddDependencies(projectJsonPath, repo.PackageDependencies);
+                        }
+                    }
+                }
+
+                repo.PackageDependencies.ExceptWith(repo.Packages);
+            }
+
+            foreach (var repo in repos)
+            {
+                foreach (var packageDependency in repo.PackageDependencies)
+                {
+                    var parentRepo = repos.Find(r => r.Packages.Contains(packageDependency));
+                    if (parentRepo != null)
+                    {
+                        parentRepo.DependentRepos.Add(repo);
+                    }
+                }
+            }
+
+            var visited = new List<Repository>();
+            foreach (var repo in repos)
+            {
+                DFSRepoGraph(repo, visited);
+            }
+        }
+
+        static void DFSRepoGraph(Repository repo, List<Repository> visited)
+        {
+            if (visited.Contains(repo))
+            {
+                throw new Exception("A cyclic dependency between the following repositories has been detected: " +
+                    string.Join(" -> ", visited));
+            }
+
+            visited.Add(repo);
+
+            foreach (var child in repo.DependentRepos)
+            {
+                child.Order = Math.Max(child.Order, repo.Order + 1);
+
+                DFSRepoGraph(child, visited);
+            }
+
+            visited.RemoveAt(visited.Count - 1);
+        }
+
+        static void AddDependencies(string projectJsonPath, HashSet<string> dependencies)
+        {
+            var root = LoadJObject(projectJsonPath);
+            dependencies.UnionWith(EnumerateDependencies(root).Select(d => d.Name));
+        }
+
+        private static Rule GetRuleForPackage(Project package, IEnumerable<Rule> rules)
+        {
+            foreach (var rule in rules)
+            {
+                if (Regex.Match(package.Name, rule.Pattern).Success &&
+                    string.Equals(package.Version.ToString(), rule.CurrentVersion, StringComparison.OrdinalIgnoreCase))
+                {
+                    return rule;
+                }
+            }
+
+            return null;
+        }
+
+        private static bool ContainsModifiedSrcProjectFile(string path)
+        {
+            var srcPath = Path.Combine(path, "src");
+            if (!Directory.Exists(srcPath))
+            {
+                return false;
+            }
+
+            var stdout = new StringBuilder();
+            var exitCode = new ProcessRunner("git", "diff -- src/**/project.json")
```

Comment:
Actually, this will probably still be needed. For example, if you update a System package, you would probably want the tool to go through every repo to figure out whether a src package needs an update instead of manually checking and then calling the tool on every repo.